landscape/client/snap_utils.py

```diff
@@ -4,6 +4,28 @@
 from landscape.client.snap_http import SnapdHttpException
 
 
+def parse_assertion(headers, signature):
+    """Parse an assertion."""
+    # monkey patch pyyaml to ignore aliases e.g. *alias when loading
+    # for instance, a system-user assertion can have the following json:
+    #   {
+    #       [...]
+    #        "system-user-authority": "*",
+    #       [...]
+    #   }
+    # which after signing gets converted to yaml that looks like:
+    #   [...]
+    #   system-user-authority: *
+    #   [...]
+    # pyyaml tries to parse the * as the start of an alias leading to errors
+    yaml.Loader.fetch_alias = yaml.Loader.fetch_plain
+    yaml.Loader.yaml_implicit_resolvers.pop("*", None)
+
+    assertion = yaml.load(headers, yaml.Loader)
```

Comment:
Just curious: why can't we use `safe_load` anymore? Is it because the above monkeypatching doesn't work the same on `yaml.SafeLoader`?

Reply:
It actually does work, good catch,... thanks